python-threatexchange/threatexchange/meta.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+A way to connect all the key interfaces of the library.
+
+Since matching may need to look up context from the original fetching,
+this is all the parts needed to get you there.
+
+You don't need to do it this way, but it does make it clear which
+fetching, matching, and other capabilities you are supporting.
+"""
+
+import typing as t
+from dataclasses import dataclass
+
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.fetcher.fetch_api import SignalExchangeAPI
+from threatexchange.fetcher.fetch_state import FetchedStateStoreBase
+from threatexchange.fetcher.collab_config import CollaborationConfigStoreBase
+
+
+class SignalTypeMapping:
+    def __init__(
+        self,
+        content_types: t.List[t.Type[ContentType]],
+        signal_types: t.List[t.Type[SignalType]],
```

Comment:
I can think of theoretical ones, but those might not be useful. The usecase I thinking of is from inside contexts where we would delibately only expose a subset of functionality to the library. I.e. inside the PDQ special-cased lamda method, the only signal type that exists is PDQ.

Reply:
Another note - this is a moot point right now because I don't think there are any types with multiple types right now.

Imagine that we made PDQ work on videos, and it looked for violating frames against the same index - I might enable PDQ for photos, but not want to process videos (for time).